### PR TITLE
[aiobotocore] generic improvements

### PR DESCRIPTION
**Pull request by palazzem**
### Overview

* removes external dependencies to execute `aiobotocore` tests
* `async with` context manager is returned as is, without yielding it's value (so it acts exactly like a proxy)
* `patch()` method signature compliant with other integrations
* refactoring `WrappedClientResponseContentProxy`
* removed custom implementation of a `moto` web server
* `aiobotocore` tests for versions 0.2, 0.3 and 0.4
* tests refactoring

### Notes
External dependencies have been removed:
* `moto` replaced with a container that runs the standalone server
* `asynctest` using the internal test suite

The `moto` docker container is available in a [custom alpine linux image][1], provided by the following commit: https://github.com/palazzem/moto/commit/577a680665286bdd163c342c486250e92cc5a736

[1]: https://hub.docker.com/r/palazzem/moto/

### CODE REVIEW

File: ddtrace/contrib/aiobotocore/__init__.py [hunk 1]
```diff
@@ -1,5 +1,5 @@
 """
-The aiootocore integration will trace all aws calls made with the aiobotocore
+The aiootocore integration will trace all AWS calls made with the `aiobotocore`
```
Comment (LotharSee) [inline ddtrace/contrib/aiobotocore/__init__.py]: Typo `aiootocore`

Comment (palazzem) [inline ddtrace/contrib/aiobotocore/__init__.py]: ouch!

File: docker-compose.yml [hunk 2]
```diff
@@ -36,6 +36,15 @@ memcached:
     image: memcached:1.4
     ports:
         - "127.0.0.1:${TEST_MEMCACHED_PORT}:11211"
+moto:
+    image: palazzem/moto:1.0.1
```
Comment (LotharSee) [inline docker-compose.yml]: Also add a link to your fork here to show where this comes from?

Comment (palazzem) [inline docker-compose.yml]: sure thing!

**Pull request merged**